shared/util/cloner.go

```diff
@@ -0,0 +1,38 @@
+package util
+
+import (
+	"encoding/json"
+	"reflect"
+)
+
+func DeepClone(dst, src interface{}) error {
+	data, err := json.Marshal(src)
+	if err != nil {
+		return err
+	}
+
+	if err = json.Unmarshal(data, &dst); err != nil {
+		return err
+	}
+	return nil
+}
```

Comment:
This might be a good use case for golang generics rather than `interface{}`. They've also updated `interface{}` to be called `any`

Reply:
Since we don't really manipulate any functions generics might expose, I'll change the parameter type to `any`.